### PR TITLE
Add Switchplate 910 keyboard

### PR DESCRIPTION
**Pull request by MxBlu**
## Description

Adding Switchplate 910 keyboard on behalf of Stefan Karsh (hence the license headers)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/switchplate/switchplate910/config.h [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+//#define MAGIC_KEY_LAYER8         8
+//#define MAGIC_KEY_LAYER9         9
+//#define MAGIC_KEY_BOOTLOADER     B
+//#define MAGIC_KEY_BOOTLOADER_ALT ESC
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_EEPROM_CLEAR   BSPACE
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
```
Comment (fauxpark) [inline keyboards/switchplate/switchplate910/config.h:117]:
```suggestion
```

File: keyboards/switchplate/switchplate910/rules.mk [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (fauxpark) [inline keyboards/switchplate/switchplate910/rules.mk:11]:
```suggestion
```

File: keyboards/switchplate/switchplate910/rules.mk [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/switchplate/switchplate910/rules.mk:30]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/switchplate/switchplate910/switchplate910.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+
+void matrix_init_kb(void) {
+    matrix_init_user();
+}
+
+bool led_update_kb(led_t led_state) {
+    return led_update_user(led_state);
+}
+
```
Comment (fauxpark) [inline keyboards/switchplate/switchplate910/switchplate910.c:32]:
```suggestion
```

File: keyboards/switchplate/switchplate910/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    KC_LCTL, KC_LGUI, KC_LALT,           KC_SPACE,           KC_RALT, MO(1), KC_LEFT, KC_DOWN, KC_RGHT
+),
+
+    LAYOUT_all( /* L1 */
+	KC_GRV, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_DEL, KC_INS, KC_TRNS,
+    KC_TRNS, KC_MPLY, KC_MPRV, KC_MNXT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_HOME, KC_PSCR, KC_TRNS, KC_TRNS, KC_TRNS, RGB_VAI,
+    RESET, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_END, KC_TRNS, KC_TRNS, KC_TRNS, RGB_VAD,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MUTE, KC_VOLD, KC_VOLU, KC_TRNS, RGB_MOD, RGB_SAI, RGB_TOG,
+    KC_TRNS, KC_TRNS, KC_TRNS,           KC_TRNS,           KC_TRNS, KC_TRNS, RGB_HUD, RGB_SAD, RGB_HUI
+),
```
Comment (fauxpark) [inline keyboards/switchplate/switchplate910/keymaps/via/keymap.c:39]: Need two extra empty layers.

File: keyboards/switchplate/switchplate910/switchplate910.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	{ K400,  K401,  K402,  KC_NO, KC_NO, K405,  KC_NO, KC_NO, K408,  K409,  KC_NO, K411,  K412,  KC_NO, K414  }  \
+}
+
+
+
+
+
+
+
+
```
Comment (fauxpark) [inline keyboards/switchplate/switchplate910/switchplate910.h:51]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by MxBlu** [comment]
Faster than I was reading through the PR checklist, done

File: keyboards/switchplate/switchplate910/rules.mk [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/switchplate/switchplate910/rules.mk:20]:
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/switchplate/switchplate910/rules.mk [hunk 8]
```diff
@@ -0,0 +1,20 @@
+# MCU name
+MCU = atmega32u4
+
+BOOTLOADER = atmel-dfu
```
Comment (fauxpark) [inline keyboards/switchplate/switchplate910/rules.mk:4]:
```suggestion
# Bootloader selection
BOOTLOADER = atmel-dfu
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/switchplate/switchplate910/rules.mk (hunk 7 above)
Comment (MxBlu) [reply keyboards/switchplate/switchplate910/rules.mk:20]: Even though the PCB design doesn't really make BT a possibility at all?

Comment (fauxpark) [reply keyboards/switchplate/switchplate910/rules.mk:21]: Yes, it's just part of the template.

Comment (MxBlu) [reply keyboards/switchplate/switchplate910/rules.mk:21]: Cool, noted.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**